Update Readme (#42)
Shopify/i18next-shopify: README.md

# `@shopify/i18next-shopify`

This package is an `i18nFormat` plugin for the [i18next](https://www.i18next.com/) library. It allows developers to use the same format used by Shopify [apps](https://shopify.dev/docs/apps/checkout/best-practices/localizing-ui-extensions#how-it-works) and [themes](https://shopify.dev/docs/themes/architecture/locales/storefront-locale-files#usage) for localization.

> Key features include

Reviewer: ```suggestion
Key features include:
```

> * String interpolation within both single and double curly brace mustache formats

Reviewer: ```suggestion
* String interpolation with both single `{}` and double `{{}}` curly brace mustache formats
```

> * Search nested locations for plural handling

Reviewer: ```suggestion
* Pluralization as nested keys
```

  * E.g. `{"keyWithCount": {"one": "value"}}` instead of `{"keyWithCount_one": "value"}`

> * Allow for the "ordinal" interpolation technique for ordinal pluralization

Reviewer: ```suggestion
* Ordinal pluralization using the interpolation variable `ordinal` as number (e.g. `t("key", { ordinal: 1 })`) or using the interpolation variable `ordinal` as boolean and `count` as number (e.g. `t("key", { ordinal: true, count: 1 })`)
```

>   * NOTE: An edge case to consider is when the `ordinal` value is explicitly set to 0 and `count` is provided. This plugin treats this as cardinal pluralization

Reviewer: ```suggestion
  * NOTE: When `ordinal` is explicitly set to `0` and `count` is also explicitly set, the plugin will treat this as cardinal pluralization (e.g. `t("key", { count: 1, ordinal: 0 }`))
```

## Table of contents
[Getting Started](#getting-started)\
[Example](#example)\
[License](#license)\
[Contributing](#contributing)\
[References](#references)

## Getting started

### Add the package
```
$ npm install @shopify/i18next-shopify
```
or
```
$ yarn add @shopify/i18next-shopify
```

> ### Configure i18next to use this plugin:

Reviewer: ```suggestion
### Configure i18next to use this plugin
```

i18next.use(ShopifyFormat).init(i18nextOptions);
```

> ## Example

Reviewer: I'm not sure where, but it would be good to put somewhere in the README that this plugin supports interpolation variables with single and double curly braces

When creating a new issue, please ensure the issue is clear and include additional details to help maintainers reproduce it:

Use a clear and descriptive title for the issue to identify the problem.
- Describe the exact steps which reproduce the problem in as many details as possible.
- Provide specific examples to demonstrate the steps. Include links to files, or copy/pasteable snippets. If you're providing snippets in the issue, use Markdown code blocks.
- Describe the behavior you observed after following the steps and point out what exactly is the problem with that behavior.
- Explain which behavior you expected to see instead and why.
- Include screenshots and animated GIFs where possible.

### Contributing Code

The following is a list of useful commands and hits.
- `yarn install` to install development dependencies
- `yarn test` to run the test suite
- `yarn tdd` to run tests after files are update
- `yarn lint` to run the linter
- `yarn changeset` to generate a changeset.

> - Open a PR with your contribution.  Fill out all applicable sections of the [PR template](https://github.com/Shopify/i18next-shopify/blob/main/.github/pull_request_template.md)

Reviewer: ```suggestion

How to contribute:
- Open a PR with your contribution.  Fill out all applicable sections of the [PR template](https://github.com/Shopify/i18next-shopify/blob/main/.github/pull_request_template.md)
```
